### PR TITLE
Expose sdk.fromWallet and wallet classes

### PR DESCRIPTION
**Pull request by adam-maj**

### CODE REVIEW

File: packages/sdk/src/evm/wallets/core/aws.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        );
+      }
+
+      this.cachedSigner = new ethers.Wallet(privateKey, provider);
+      return this.cachedSigner;
+    } catch (error) {
+      // For a list of exceptions thrown, see
+      // https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html
+      throw error;
+    }
```
Comment (jnsdls) [inline packages/sdk/src/evm/wallets/core/aws.ts:83]: assuming we'll add handling of some sort here? If not why even wrap this all in a try catch in the first place?

File: packages/sdk/src/evm/core/sdk.ts [hunk 2]
```diff
@@ -39,6 +40,48 @@ import invariant from "tiny-invariant";
...
+   * });
+   * const sdk = await ThirdwebSDK.fromWallet(wallet, "mainnet");
+   * ```
+   *
+   * @param wallet - the implementation of the AbstractWallet class to use for signing
+   * @param network - the network (chain) to connect to (e.g. "mainnet", "rinkeby", "polygon", "mumbai"...) or a fully formed RPC url
+   * @param options - the SDK options to use
+   * @returns an instance of the SDK
+   * @param storage - optional storage implementation to use
+   */
```
Comment (jnsdls) [inline packages/sdk/src/evm/core/sdk.ts:66]: lets mark this `@beta` for now

Comment (jnsdls) [reply packages/sdk/src/evm/core/sdk.ts:66]: as in the entire fn, also let's group the params have @returns below

**Review by jnsdls — CHANGES_REQUESTED** [review]
couple of changes, couple of questions that we should think about and discuss

File: packages/sdk/src/evm/wallets/core/aws.ts (hunk 1 above)
Comment (adam-maj) [reply packages/sdk/src/evm/wallets/core/aws.ts:83]: Might as well remove - just AWS default copy-paste

File: packages/sdk/src/evm/wallets/core/async.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    this.getPrivateKey = options.getPrivateKey;
+  }
+
+  async getSigner(
+    provider?: ethers.providers.Provider,
+  ): Promise<ethers.Signer> {
+    const privateKey = await this.getPrivateKey();
+    return new ethers.Wallet(privateKey, provider);
+  }
+}
```
[the remark this answers is not in the document]
Comment (adam-maj) [reply packages/sdk/src/evm/wallets/core/async.ts:41]: This one would be for all the other cloud providers/async ways to get PK that we should support - but we aren't rolling out support for everything initially - we should probably have something like that so people aren't just locked into using AWS for this (or having to write their own wallet implementation)

[the remark this answers is not in the document]
Comment (jnsdls) [reply packages/sdk/src/evm/wallets/core/async.ts:41]: I guess how is this different than just proving the AbstractWallet and letting them implement on top of that?

File: packages/sdk/src/evm/wallets/core/aws.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+ * ```
+ */
+export class AWSSecretsManagerWallet extends AbstractWallet {
+  private secretId: string;
+  private secretKeyName: string;
+  private client: SecretsManagerClient;
+  private cachedSigner: ethers.Signer | undefined;
+
+  constructor(options: AwsSecretsManagerWalletOptions) {
+    super();
```
Comment (jnsdls) [inline packages/sdk/src/evm/wallets/core/aws.ts:45]: does `AbstractWallet` have a constructor? if not this is not needed

**Review by jnsdls — COMMENTED** [review]
abstract wallet doesn't need to be called with super I believe?

File: packages/sdk/src/evm/wallets/core/aws.ts (hunk 4 above)
Comment (adam-maj) [reply packages/sdk/src/evm/wallets/core/aws.ts:45]: No but it still yells at you to always call super for some reason

**Review by jnsdls — APPROVED** [review]
fix build etc then good to go

**Pull request merged**